### PR TITLE
Update to 1.20.2

### PR DESCRIPTION
**Pull request by PiTheGuy**
JAMMER didn't seem to help that much in terms of file count...

### CODE REVIEW

**Comment by sciwhiz12** [comment]
oh lawd

**Comment by Matyrobbrt** [comment]
Oh, lovely, this will be fun

**Comment by Lanse505** [comment]
I demand a raise >.>

File: data/net/minecraft/commands/CommandSourceStack.mapping [hunk 1]
```diff
@@ -82,7 +80,8 @@ CLASS net/minecraft/commands/CommandSourceStack
 		ARG 1 returnValueConsumer
 	METHOD withRotation (Lnet/minecraft/world/phys/Vec2;)Lnet/minecraft/commands/CommandSourceStack;
 		ARG 1 rotation
-	METHOD withSigningContext (Lnet/minecraft/commands/CommandSigningContext;)Lnet/minecraft/commands/CommandSourceStack;
+	METHOD withSigningContext (Lnet/minecraft/commands/CommandSigningContext;Lnet/minecraft/util/TaskChainer;)Lnet/minecraft/commands/CommandSourceStack;
 		ARG 1 signingContext
+		ARG 2 chatMessageContainer
```
Comment (sciwhiz12) [inline data/net/minecraft/commands/CommandSourceStack.mapping]: This should be `chatMessageChainer`.

```suggestion
		ARG 2 chatMessageChainer
```

File: data/net/minecraft/advancements/critereon/EntityEquipmentPredicate.mapping [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
-		METHOD legs (Lnet/minecraft/advancements/critereon/ItemPredicate;)Lnet/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder;
+		METHOD legs (Lnet/minecraft/advancements/critereon/ItemPredicate$Builder;)Lnet/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder;
 			ARG 1 legs
-		METHOD mainhand (Lnet/minecraft/advancements/critereon/ItemPredicate;)Lnet/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder;
+		METHOD mainhand (Lnet/minecraft/advancements/critereon/ItemPredicate$Builder;)Lnet/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder;
 			ARG 1 mainhand
-		METHOD offhand (Lnet/minecraft/advancements/critereon/ItemPredicate;)Lnet/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder;
-			ARG 1 offhand
+		METHOD offhand (Lnet/minecraft/advancements/critereon/ItemPredicate$Builder;)Lnet/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder;
+			ARG 1 offfhand
```
Comment (sciwhiz12) [inline data/net/minecraft/advancements/critereon/EntityEquipmentPredicate.mapping]: Typo: `offfhand` to `offhand`.
```suggestion
			ARG 1 offhand
```

File: data/net/minecraft/advancements/critereon/RecipeCraftedTrigger.mapping [hunk 3]
```diff
@@ -4,13 +4,13 @@ CLASS net/minecraft/advancements/critereon/RecipeCraftedTrigger
 		ARG 2 recipeId
 		ARG 3 items
 	CLASS TriggerInstance
-		METHOD <init> (Lnet/minecraft/advancements/critereon/ContextAwarePredicate;Lnet/minecraft/resources/ResourceLocation;Ljava/util/List;)V
+		METHOD <init> (Ljava/util/Optional;Lnet/minecraft/resources/ResourceLocation;Ljava/util/List;)V
 			ARG 1 player
-			ARG 2 recipeId
+			ARG 2 recipedId
```
Comment (sciwhiz12) [inline data/net/minecraft/advancements/critereon/RecipeCraftedTrigger.mapping]: Typo: `recipedId` to `recipeId`.
```suggestion
			ARG 2 recipeId
```

File: data/net/minecraft/client/multiplayer/ClientCommonPacketListenerImpl.mapping [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	METHOD send (Lnet/minecraft/network/protocol/Packet;)V
+		ARG 1 packet
+	METHOD send (Lnet/minecraft/network/protocol/common/ServerboundResourcePackPacket$Action;)V
+		ARG 1 action
+	METHOD sendWhen (Lnet/minecraft/network/protocol/Packet;Ljava/util/function/BooleanSupplier;Ljava/time/Duration;)V
+		ARG 1 packet
+		ARG 2 sendCondition
+		ARG 3 expirationTime
+	METHOD showServerPackPrompt (Ljava/net/URL;Ljava/lang/String;ZLnet/minecraft/network/chat/Component;)V
+		ARG 1 resoursePackUrl
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/ClientCommonPacketListenerImpl.mapping]: Typo: `resoursePackUrl` to `resourcePackUrl`.

```suggestion
		ARG 1 resourcePackUrl
```

File: data/net/minecraft/world/effect/MobEffect.mapping [hunk 5]
```diff
@@ -10,10 +10,9 @@ CLASS net/minecraft/world/effect/MobEffect
 		ARG 2 uuid
 		ARG 3 amount
 		ARG 5 operation
-	METHOD addAttributeModifiers (Lnet/minecraft/world/entity/LivingEntity;Lnet/minecraft/world/entity/ai/attributes/AttributeMap;I)V
-		ARG 1 livingEntity
-		ARG 2 attributeMap
-		ARG 3 amplifier
+	METHOD addAttributeModifiers (Lnet/minecraft/world/entity/ai/attributes/AttributeMap;I)V
+		ARG 1 attrubuteMap
```
Comment (sciwhiz12) [inline data/net/minecraft/world/effect/MobEffect.mapping]: Typo: `attrubuteMap` to `attributeMap`.

```suggestion
		ARG 1 attributeMap
```

File: data/net/minecraft/world/level/storage/loot/predicates/WeatherCheck.mapping [hunk 6]
```diff
@@ -1,13 +1,10 @@
...
 	METHOD test (Lnet/minecraft/world/level/storage/loot/LootContext;)Z
 		ARG 1 context
 	CLASS Builder
-		METHOD setRaining (Ljava/lang/Boolean;)Lnet/minecraft/world/level/storage/loot/predicates/WeatherCheck$Builder;
+		METHOD setRaining (Z)Lnet/minecraft/world/level/storage/loot/predicates/WeatherCheck$Builder;
 			ARG 1 isRaining
-		METHOD setThundering (Ljava/lang/Boolean;)Lnet/minecraft/world/level/storage/loot/predicates/WeatherCheck$Builder;
-			ARG 1 isThundering
+		METHOD setThundering (Z)Lnet/minecraft/world/level/storage/loot/predicates/WeatherCheck$Builder;
+			ARG 1 isThudering
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/storage/loot/predicates/WeatherCheck.mapping]: Typo: `isThudering` to `isThundering`.

```suggestion
			ARG 1 isThundering
```

File: data/net/minecraft/world/level/storage/loot/entries/LootTableReference.mapping [hunk 7]
```diff
@@ -1,24 +1,10 @@
 CLASS net/minecraft/world/level/storage/loot/entries/LootTableReference
 	COMMENT A loot pool entry container that generates loot by referencing another loot table.
-	METHOD <init> (Lnet/minecraft/resources/ResourceLocation;II[Lnet/minecraft/world/level/storage/loot/predicates/LootItemCondition;[Lnet/minecraft/world/level/storage/loot/functions/LootItemFunction;)V
-		ARG 1 lootTableId
+	METHOD <init> (Lnet/minecraft/resources/ResourceLocation;IILjava/util/List;Ljava/util/List;)V
+		ARG 1 name
 		ARG 2 weight
 		ARG 3 quality
-		ARG 4 conditions
+		ARG 4 coditions
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/storage/loot/entries/LootTableReference.mapping]: Typo: `coditions` to `conditions`.

```suggestion
		ARG 4 conditions
```

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Skimmed the PR, mostly looking for typos and obvious errors.

Didn't go too indepth with comparing between the code and the names, so there _might_ be some things that'll slip by us. I don't think that'll be too much of an issue.

I am now contemplating making a script which collects all of the parameter names and does word-comparisons to find likely typos; as in, check how similar two words are, and log if they're very similar and the change isn't something like an `s` appended to the end. Probably will be an external script, not included in this repo (unless I think it's good enough to be included).

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**